File: Boot/Brasserie/src/main/java/soprajc/Brasserie/model/Brasseur.java
package soprajc.Brasserie.model;

import java.util.List;
import java.util.Objects;

import javax.persistence.Entity;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@Table(name="brasseur")
public class Brasseur extends Compte{

	@JsonView(JsonViews.Common.class)
	private double recettes;
	
	@JsonView(JsonViews.Common.class)
	private double depenses;
	
	@JsonView(JsonViews.Common.class)
	private double tresorerie;
	
	// constructors
	public Brasseur() {}
	
	public Brasseur(String mail, String password, double recettes,
			double depenses, double tresorerie) {
		super(mail, password, null);
		this.recettes = recettes;
		this.depenses = depenses;
		this.tresorerie = tresorerie;
	}

	public Brasseur(String mail, String password) {
		super(mail, password);
	}

	// getters-setters
	public double getRecettes() {
		return recettes;
	}
	public void setRecettes(double recettes) {
		this.recettes = recettes;
	}

	public double getDepenses() {
		return depenses;
	}
	public void setDepenses(double depenses) {
		this.depenses = depenses;
	}

	public double getTresorerie() {
		return tresorerie;
	}
	public void setTresorerie(double tresorerie) {
		this.tresorerie = tresorerie;
	}
	
	// methods
	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Brasseur other = (Brasseur) obj;
		return Objects.equals(id, other.id);
	}
	
	public List<Biere> recommandation(Client c){
		return null;
	}
	
	public void preparerCommande(Client c) {
		
	}
	
}

File: Angular/Brasserie/src/app/model/brasseur.ts
import { Compte } from './compte';

export class Brasseur extends Compte {
  public constructor(
    _id?: number | undefined,
    _mail?: String | undefined,
    _password?: String | undefined,
    private _recettes?: number | undefined,
    private _depenses?: number | undefined,
    private _tresorerie?: number | undefined
  ) {
    super(_id, _mail, _password);
  }

  public get recettes(): number | undefined {
    return this._recettes;
  }

  public set recettes(value: number | undefined) {
    this._recettes = value;
  }

  public get depenses(): number | undefined {
    return this._depenses;
  }

  public set depenses(value: number | undefined) {
    this._depenses = value;
  }

  public get tresorerie(): number | undefined {
    return this._tresorerie;
  }

  public set tresoreri(value: number | undefined) {
    this._tresorerie = value;
  }
}
